ddtrace/tracer.py

```diff
@@ -387,20 +395,13 @@ def start_span(self, name, child_of=None, service=None, resource=None, span_type
 
         # Add env, service, and version tags
         # DEV: These override the default global tags, `DD_VERSION` takes precedence over `DD_TAGS=version:v`
-        dd_tags = {}
         if config.env:
-            dd_tags[ENV_KEY] = config.env
-        # TODO: Only set this if `service` == `config.service` (`DD_SERVICE`)
-        if config.version:
-            dd_tags[VERSION_KEY] = config.version
-        span.set_tags(dd_tags)
+            span.set_tag(ENV_KEY, config.env)
+        if service == config.service and config.version:
```

Comment:
Do we have a test for this condition?

Reply:
not currently, but we should!

Reply:
Also I'm still not 100% on having the condition `service == config.service` in order to set the version tag. @brettlangdon mention here https://github.com/DataDog/dd-trace-py/pull/1211#discussion_r391207313
> we want to make sure the version only relates to the primary service for the app.

which makes sense. But I'm concerned about a user setting `config.version`/`DD_VERSION` and not having it work because they didn't set `config.service`/`DD_SERVICE`. This could happen if they set a span service in some custom instrumentation (not through the `config` api).

Are we okay with just documenting this (which we'll need to do), or do we want to try to come up with a solution for this case?

Reply:
this probably needs to be... only set `version` if the service name matches the root spans service name.

so if someone is only using the default service names (e.g. `config.service` is not set), then we still want this to work.

Reply:
Yep, this makes sense. We'll use the root span.